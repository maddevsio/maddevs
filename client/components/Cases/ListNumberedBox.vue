<template>
  <ol class="case-list_numbered">
    <slot></slot>
  </ol>
</template>

<script>
export default {
  name: 'ListNumberedBox',
}
</script>

<style scoped lang="scss">
@import '../../assets/styles/cases/_mixins';

.case-list_numbered {
  padding: 0;
  counter-reset: section;
  list-style-type: none;
  display: flex;
  flex-direction: column;
}
</style>
