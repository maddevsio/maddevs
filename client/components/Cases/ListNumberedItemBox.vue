<template>
  <li class="case_list-numbered_item">
    <slot></slot>
  </li>
</template>

<script>
export default {
  name: 'ListNumberedItemBox',
}
</script>

<style scoped lang="scss">
@import '../../assets/styles/cases/_mixins';

.case_list-numbered_item {
  @include list-item_box($bgcolor--silver, 16px 17px 16px 40px, 8px);
  @include default_text($text-color--black-cases, 17px, 166%, -0.035em, normal);
  list-style-type: none;

  &::before {
    counter-increment: section;
    content: counter(section) '.';
    position: absolute;
    left: 16px;
  }

  @media screen and (max-width: 768px) {
    @include list-item_box($bgcolor--silver, 8px 16px 8px 40px, 8px);
    @include default_text($text-color--black-cases, 16px, 150%, -0.02em, normal);
    margin-bottom: 10px;
  }
}
</style>
