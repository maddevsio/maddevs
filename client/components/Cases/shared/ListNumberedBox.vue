<template>
  <ol class="case-list_numbered">
    <slot />
  </ol>
</template>

<script>
export default {
  name: 'ListNumberedBox',
}
</script>

<style scoped lang="scss">
.case-list_numbered {
  padding: 0;
  counter-reset: section;
  list-style-type: none;
  display: flex;
  flex-direction: column;
}
</style>
