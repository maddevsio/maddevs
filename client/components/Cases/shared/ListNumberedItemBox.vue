<template>
  <li class="case_list-numbered_item">
    <slot />
  </li>
</template>

<script>
export default {
  name: 'ListNumberedItemBox',
}
</script>

<style scoped lang="scss">
.case_list-numbered_item {
  @include list-item-box($bgcolor--silver, 16px 17px 16px 40px, 8px);
  @include default-text($text-color--black-oil, 17px, 166%, -0.035em, normal);
  list-style-type: none;

  &::before {
    counter-increment: section;
    content: counter(section) '.';
    position: absolute;
    left: 16px;
  }

  @media screen and (max-width: 768px) {
    @include list-item-box($bgcolor--silver, 8px 16px 8px 40px, 8px);
    @include default-text($text-color--black-oil, 16px, 150%, -0.02em, normal);
    margin-bottom: 10px;
  }
}
</style>
