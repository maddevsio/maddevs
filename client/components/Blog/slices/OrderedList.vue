<template>
  <list-numbered-box>
    <list-numbered-item-box v-for="(item, index) in slice.items" :key="index">
      {{$prismic.asText(item.list_item)}}
    </list-numbered-item-box>
  </list-numbered-box>
</template>

<script>
import ListNumberedBox from '@/components/Cases/ListNumberedBox';
import ListNumberedItemBox from '@/components/Cases/ListNumberedItemBox';
export default {
  name: 'OrderedList',
  components: {
    ListNumberedItemBox,
    ListNumberedBox
  },
  props: {
    slice: {
      type: Object,
      required: true
    }
  }
};
</script>
