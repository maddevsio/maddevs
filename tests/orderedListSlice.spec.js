import {mount} from '@vue/test-utils';
import OrderedList from '../client/components/Blog/slices/OrderedList';
import ListNumberedBox from '../client/components/Cases/ListNumberedBox';
import ListNumberedItemBox from '../client/components/Cases/ListNumberedItemBox';

describe('OrderedListSlice component', () => {
  let wrapper;

  const slice = {
    items: [
      {list_item: 'list item 1'},
      {list_item: 'list item 2'},
      {list_item: 'list item 3'},
      {list_item: 'list item 4'},
      {list_item: 'list item 5'}
    ]
  };

  beforeEach(() => {
    wrapper = mount(OrderedList, {
      mocks: {
        $prismic: {
          asText: text => text
        }
      },
      propsData: {slice}
    });
  });

  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('contains list box components', () => {
    expect(wrapper.findComponent(ListNumberedBox).exists()).toBeTruthy();
    expect(wrapper.findComponent(ListNumberedItemBox).exists()).toBeTruthy();
  });
});
