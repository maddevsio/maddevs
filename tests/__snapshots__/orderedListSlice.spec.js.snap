// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderedListSlice component renders correctly 1`] = `
<ol
  class="case-list_numbered"
>
  <li
    class="case_list-numbered_item"
  >
    
    list item 1
  
  </li>
  <li
    class="case_list-numbered_item"
  >
    
    list item 2
  
  </li>
  <li
    class="case_list-numbered_item"
  >
    
    list item 3
  
  </li>
  <li
    class="case_list-numbered_item"
  >
    
    list item 4
  
  </li>
  <li
    class="case_list-numbered_item"
  >
    
    list item 5
  
  </li>
</ol>
`;
